### ISSUE TITLE
Wallet - Endless spinner appears on the ‘Save’ button when adding a debit card

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#34823]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.28-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com or open App
2. Login with any account
3. Go to Settings > Wallet
4. Click/Tap on Add bank account > Debit card
5. Fill in all fields with correct values
6. Click/Tap on 'Save' button

## Expected Result:
Debit card added successfully

## Actual Result:
Endless spinner appears on the ‘Save’ button when adding a debit card

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/a653f25d-f4c3-4c26-b36c-ac335e413fd1

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by mountiny** [comment]
I dont see any changes related to this. @lanitochka17 could you should Inspector and the networks tab so we can see the network request being called in this case?

**Comment by bondydaa** [comment]
hmm seems this is reproducible on staging for me 

<img width="1926" alt="image" src="https://github.com/Expensify/App/assets/4073354/09b55670-a7d1-4bf1-9075-f6bba1143a77">

The request for `AddPaymentCard` doesn't seem to return, network inspector doesn't show anything for the request.

Oh looks like it finally timed out after ~3minutes
<img width="1940" alt="image" src="https://github.com/Expensify/App/assets/4073354/b41f8f5d-feb6-4849-9f74-4a66eb91720f">

Looks like we did actually make a request, logs https://www.expensify.com/_devportal/tools/logSearch/#query=request_id%3A%228482beb3ae9a51dd-DEN%22%20AND%20timestamp%3A%5B2024-01-19T00%3A00%20TO%202024-01-20T23%3A59%5D&index=_all

I'm not exactly sure what happened there though 🤔

**Comment by bondydaa** [comment]
OH I wonder if this is b/c of some networking changes we're doing on the backend, it seems like we fail pretty quickly when we try to talk to bancorp so my hunch is that we maybe haven't updated to whitelist our new IPs with bancorp (or at least I hope it's that simple 🙏 ) 

I don't see anything from the past 2 deploy checklists that stands out as a cause.

**Comment by bondydaa** [comment]
bumped our internal only thread there asking for help to figure out if this is b/c of our new IPs https://expensify.slack.com/archives/C094TGUTZ/p1705707029479859?thread_ts=1705594947.763029&cid=C094TGUTZ

**Comment by mvtglobally** [comment]
Additional Network information 

![image](https://github.com/Expensify/App/assets/43995119/5d76467d-3983-4e1e-9ff3-ad75dbc08fd2)
https://github.com/Expensify/App/assets/43995119/6dd6a5cf-cdc3-4b67-98e5-a734acf6eea9

**Comment by bondydaa** [comment]
I'll leave the blocker on here for now but I'm pretty sure this isn't a new dot blocker and has to do with some internal networking changes we made, we're working on clarifying with the rest of the team so if the new dot deployer on monday still sees this and needs to get something out I think they can remove it unless someone is able to confirm it is something wrong in new dot.

**Comment by mountiny** [comment]
As discussed above this is not App blocker, no code in the staging diff is related to this and seems like there is a culprit investigated in the backend

**Comment by bondydaa** [comment]
yep we got confirmation finally from bancorp they didn't properly update some of their firewalls with our new IPs. I'm going to demote this to weekly as they should get back to us this week and we can retest and confirm no other issues.

**Comment by bondydaa** [comment]
our contacts said the network configurations should be fixed last night, testing again... and still getting spinners so don't think it's fully fixed on their end yet.

**Comment by bondydaa** [comment]
okay this is still an issue but it's an internal problem with talking to our partner's servers and IP whitelisting. I'm going to transfer this issue to expensify/expensify b/c it's an internal issue and we need to keep working with those partners.
